llvm/lib/SYCLLowerIR/ModuleSplitter.cpp

```diff
@@ -733,6 +737,14 @@ void EntryPointGroup::rebuild(const Module &M) {
       Functions.insert(const_cast<Function *>(&F));
 }
 
+std::string ModuleDesc::makeSymbolTable() const {
+  std::string ST;
```

Comment:
[`llvm::SmallString`](https://llvm.org/doxygen/classllvm_1_1SmallString.html) would be a better fit to reduce amount of re-allocations

Reply:
The total size of a string is rarely small to benefit from small string optimizations. C++ mangled names are very long. What size would you suggest for SmallString for this case?

Reply:
It is not that it would fit into a pre-allocated area on the stack, it is that `SmallString` is a wrapper around `SmallVector`, which does not fully re-allocate on every `+=` operation, because its capacity grows at a different pace similar to `std::vector`'s `push_back`